code/game/objects/items/weapons/ecigs.dm

```diff
@@ -0,0 +1,322 @@
+				to_chat(user, SPAN_NOTICE("You can't use \the [src] with no cartridge installed!"))
+				return
+			else if(!ec_cartridge.reagents.total_volume)
+				to_chat(user, SPAN_NOTICE("You can't use \the [src] with no liquid left!"))
+				return
+			else if(!cigcell.check_charge(power_usage * CELLRATE))
+				to_chat(user, SPAN_NOTICE("\The [src]'s power meter flashes a low battery warning and refuses to operate."))
+				return
+			active = TRUE
+			START_PROCESSING(SSprocessing, src)
+			to_chat(user, SPAN_NOTICE("You turn on \the [src]."))
+			update_icon()
+
+		else
+			to_chat(user, SPAN_WARNING("\The [src] does not have a battery installed."))
+
+/obj/item/clothing/mask/smokable/ecig/attack_hand(mob/user)//eject cartridge
+	if(user.get_inactive_hand() == src)//if being hold
+		if (ec_cartridge)
+			active = FALSE
+			user.put_in_hands(ec_cartridge)
+			to_chat(user, SPAN_NOTICE("You remove \the [ec_cartridge] from \the [src]."))
+			ec_cartridge = null
+			update_icon()
+	else
+		..()
+
+/obj/item/clothing/mask/smokable/ecig/attack(mob/living/carbon/human/C, mob/user, def_zone)
+	if(active && C == user && istype(C))
+		var/obj/item/blocked = C.check_mouth_coverage()
+		if(blocked)
+			to_chat(C, SPAN_WARNING("\The [blocked] is in the way!"))
+			return TRUE
+		if(last_drag <= world.time - 30) 
+			if(!cigcell.checked_use(power_usage * CELLRATE)) //if this passes, there's not enough power in the battery
+				Deactivate()
+				to_chat(C,SPAN_NOTICE("\The [src]'s power meter flashes a low battery warning and shuts down."))
+				return
+			last_drag = world.time
+			idle = 0
+			C.visible_message(SPAN_NOTICE("[C.name] takes a drag of their [name]."))
+			playsound(C, 'sound/items/cigs_lighters/inhale.ogg', 50, 0, -1)
+			ec_cartridge.reagents.trans_to_mob(C, REM, CHEM_BREATHE, 0.4) 
+			return TRUE
+	return ..()
+
+/obj/item/reagent_containers/ecig_cartridge
+	name = "tobacco flavour cartridge"
+	desc = "A small metal cartridge, used with electronic cigarettes, which contains an atomizing coil and a solution to be atomized."
+	w_class = ITEMSIZE_TINY
+	icon = 'icons/obj/ecig.dmi'
+	icon_state = "ecartridge"
+	matter = list(MATERIAL_ALUMINIUM = 50, MATERIAL_GLASS = 10)
+	volume = 20
+	flags = OPENCONTAINER
+
+/obj/item/reagent_containers/ecig_cartridge/examine(mob/user)//to see how much left
+	. = ..()
+	to_chat(user, "The cartridge has [reagents.total_volume] units of liquid remaining.")
+
+//flavours
+/obj/item/reagent_containers/ecig_cartridge/blank
+	name = "ecigarette cartridge"
+	desc = "A small metal cartridge which contains an atomizing coil."
+
+/obj/item/reagent_containers/ecig_cartridge/blanknico
+	name = "flavorless nicotine cartridge"
+	desc = "A small metal cartridge which contains an atomizing coil and a solution to be atomized. The label says you can add whatever flavoring agents you want."
+/obj/item/reagent_containers/ecig_cartridge/blanknico/New()
+	..()
+	reagents.add_reagent(/decl/reagent/toxin/tobacco/liquid, 5)
```

Comment:
i think there's a list you can use, reagents_to_add, so you don't need to add it in initialize, check `syringes.dm`